utils/TaskFileLoader.py

```diff
@@ -16,15 +17,23 @@ def LoadSparseFromTask(infilename, inputCfg):
 
     Returns
     ----------
-    - list of sparses with reconstructed quantities for all candidates, and prompt, FD D mesons and bkg candidates (only if MC)
+    - list of sparses with reconstructed quantities for all candidates,
+      and prompt, FD D mesons and bkg candidates (only if MC)
     - list of sparses with generated quantities for prompt and FD D mesons (only if MC)
     '''
     print('Loading THnSparses from file', infilename)
     infileData = TFile(infilename)
     indirData = infileData.Get(inputCfg['dirname'])
     inlistData = indirData.Get(inputCfg['listname'])
+
+    if inputCfg['sparsenameAll'] == inputCfg['sparsenamePrompt'] or \
+        inputCfg['sparsenameAll'] == inputCfg['sparsenameFD']:
+        print('ERROR: do not use the same object for different spareses, this gives an error when merged! Exit')
+        sys.exit()
+
     sparses, sparsesGen = {}, {}
-    sparses['RecoAll'] = inlistData.FindObject(inputCfg['sparsenameAll'])
+    if inputCfg['sparsenameAll'] and inlistData.FindObject(inputCfg['sparsenameAll']):
```

Comment:
This check for the `FindObject` is not done for the other histograms. We can either have a check for all or none

Reply:
Done for all in b36924e